### ISSUE TITLE
Question: Refactor library with hooks

### ISSUE THREAD

**Issue by jaybe78** [faceyspacey/react-universal-component#201]
Hi @ScriptedAlchemy,

Any plan to use hooks instead of HOCs in the library ?

**Comment by ScriptedAlchemy** [comment]
Yeah, I'm working on another project which is very similar. It uses hooks, once I finalize it ill backport the mechanism to RUC

**Comment by jaybe78** [comment]
@ScriptedAlchemy I'll be more than happy to contribute if you need help

**Comment by ScriptedAlchemy** [comment]
I wouldn’t mind some assistance giving it updates. So if you’re keen I’ll gladly accept a pr!

**Comment by ScriptedAlchemy** [comment]
Moving to hooks is tricky because the class based system lets me execute static’s ahead of time.

**Comment by jaybe78** [comment]
Do preload and prelodWeak really matters ?
The docs says that those statics can be called if there's a likelihood modules will show up soon.
To be honest I've not used those in my project, I'm mainly interested in code splitting and requiring module asynchronously. That's only from my point of view though, but I don't think that prevents us from building from scratch a new implementation of react universal.

**Comment by ScriptedAlchemy** [comment]
Those were used previously but no they are no popular features. It could be possible to further upgrade. If you’re interested in getting something started. I’ll likely pick up the PR and help. Either way I need the suspense part. I’ve got a hooks based async component in webpack-external-import. Take a look at the src/react file. That might be able to be ported over
